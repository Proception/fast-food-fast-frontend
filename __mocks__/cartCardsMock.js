import yam from '../assets/images/yam.jpg';
import rice from '../assets/images/rice.jpg';
import fufu from '../assets/images/fufu.jpg';
import { removeItemFromCart, updateCartTotal, updateItemQuantity, makeOrder } from '../src/actions/orderActions';

const props = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: (e) => {},
  onChange: () => {},
  order: {
    cart: [
      {
        "id": 1,
        "imageUrl": yam,
        "price": 500,
        "title": "Boiled Yam",
      },
      {
        "id": 2,
        "imageUrl": rice,
        "price": 900,
        "title": "Rice",
      },
      {
        "id": 3,
        "imageUrl": fufu,
        "price": 500,
        "title": "Fufu",
      },
    ],
    cartTotalPrice: 0
  },
  auth: {
    isLoading: true,
    user: {
      id: 1
    }
  },
  removeItemFromCart,
  updateCartTotal,
  updateItemQuantity,
  makeOrder
};

const emptyCartProps = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: (e) => {},
  onChange: () => {},
  order: {
    cart: [],
    cartTotalPrice: 0
  },
  auth: {
    isLoading: false,
    user: {
      id: 1
    }
  },
  removeItemFromCart,
  updateCartTotal,
  updateItemQuantity,
  makeOrder
};
export { props, emptyCartProps };
  