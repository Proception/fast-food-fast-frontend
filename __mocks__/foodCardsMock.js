import yam from '../assets/images/yam.jpg';
import rice from '../assets/images/rice.jpg';
import fufu from '../assets/images/fufu.jpg';

const props = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: () => {},
  onChange: () => {}
};
export default props;
  