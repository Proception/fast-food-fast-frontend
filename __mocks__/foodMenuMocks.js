import { addItemToCart, updateCartTotal } from '../src/actions/orderActions';

const props = {
    addItemToCart,
    updateCartTotal
    };
const emptyOrderProps = {
    updateCartTotal,
    addItemToCart
  };
export { props, emptyOrderProps };