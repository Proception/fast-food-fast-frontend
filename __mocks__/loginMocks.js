import { loginUser } from '../src/actions/authActions'

const props = {
    loginUser,
    auth: {
        isLoading: false,
    },
  };
  const propsIsLoading = {
    loginUser,
    auth: {
        isLoading: true,
    },
  };
  const state = {
    loginData: {
        email: 'omasan.esimaje',
        password: 'pass',
    }
  };
export { props, state, propsIsLoading };

