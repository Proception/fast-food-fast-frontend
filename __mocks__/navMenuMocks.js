import { setToken } from '../src/actions/authActions';

const stateGuest = {
    data: [],
    auth: {
        isAuthenticated: false,
    }
  };
  const stateUser = {
    data: [],
    auth: {
        isAuthenticated: true,
        user: {
          email: "omasan.esimaje@gmail.com",
          // full_name: "benedict"
      }
    },
    setToken: setToken,
    order: {
      cart: [1,2]
    }
  };
  const stateUserEmptyCart = {
    data: [],
    auth: {
        isAuthenticated: true,
        user: {
          email: "omasan.esimaje@gmail.com",
          // full_name: "benedict"
          role_id: 200
      }
    },
    setToken: setToken,
    order: {
      cart: []
    }
  };
  const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZnVsbF9uYW1lIjoiQmVuZWRpY3QiLCJyb2xlX2lkIjoyMDAsImlhdCI6MTU1MTg3NzMyNSwiZXhwIjoxNTUxOTYzNzI1fQ.DwFFvZskGl9JeBnLD1CCnBSpYA1i0KgK1btUY1eLc9c'
  export { stateGuest, stateUser, token, stateUserEmptyCart };