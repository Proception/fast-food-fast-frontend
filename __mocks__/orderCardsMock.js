import { getOrders, getUserOrders } from '../src/actions/orderActions';
const props = {
    orders: [
        {
          "id": 1,
          "orderNo": "12",
          "price": 5000,
          "orderDate": "30 August, 2016",
          "orderStatus": "Pending",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 2,
          "orderNo": "54",
          "price": 1200,
          "orderDate": "Boiled Yam",
          "orderStatus": "In Progress",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 3,
          "orderNo": "67",
          "price": 1500,
          "orderDate": "30 October, 2017",
          "orderStatus": "Fulfilled",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 4,
          "orderNo": "33",
          "price": 3000,
          "orderDate": "30 January, 2018",
          "orderStatus": "Completed",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
      ],
    order: {
      orders: [{
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },],
      userOrders: [{
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },]
    },
    getOrders,
    getUserOrders
    };
const emptyOrderProps = {
  orders: [
      {
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 3,
        "orderNo": "67",
        "price": 1500,
        "orderDate": "30 October, 2017",
        "orderStatus": "Fulfilled",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 4,
        "orderNo": "33",
        "price": 3000,
        "orderDate": "30 January, 2018",
        "orderStatus": "Completed",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
    ],
  order: {
    orders: [],
    userOrders: [],
  },
  getOrders,
  getUserOrders
  };
export { props, emptyOrderProps };