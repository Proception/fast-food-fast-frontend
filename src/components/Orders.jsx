import React, { Component } from 'react';
import OrderCards from './OrderCards.jsx';
import OrderControls from './OrderControls.jsx';

export default class Orders extends Component {
  state = {
    orders: [
      {
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 3,
        "orderNo": "67",
        "price": 1500,
        "orderDate": "30 October, 2017",
        "orderStatus": "Fulfilled",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 4,
        "orderNo": "33",
        "price": 3000,
        "orderDate": "30 January, 2018",
        "orderStatus": "Completed",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
    ]
  };
  generateOrders(){
    const { orders } = this.state;
    return <OrderCards orders={orders}/>
  }
  render() {
    return (
      <div>
        <OrderControls/>
        <section class="order">
          <div id="sec" class="container">
            {
              this.generateOrders()
            }
            <div class="pagination">
              <a href="#" >&laquo;</a>
              <a href="#" class="active">1</a>
              <a href="#">2</a>
              <a href="#">&raquo;</a>
            </div>
          </div>
        </section>
      </div>
    )
  }
}
