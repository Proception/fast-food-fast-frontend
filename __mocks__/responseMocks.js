import { token } from './navMenuMocks';

export const okResponse = {
    status: 200,
    response: {
        code: 200,
        status: 'ok',
        messages: 'User logged in successfully',
        data: token
    }
};

export const unauthorizedResponse = {
    status: 401,
    response: {
        code: 401,
        status: 'ok',
        messages: 'User is not authorized',
        data: token
    }
};

export const createdResponse = {
    status: 201,
    response: {
        code: 201,
        status: 'Created',
        messages: 'User signed up in successfully',
        data: token
    }
};

export const noContentResponse = {
    status: 204,
    response: {
        code: 204,
        status: 'Created',
        messages: 'User signed up in successfully',
        data: token
    }
};

export const conflictResponse = {
    status: 409,
    response: {
        code: 409,
        status: 'Conflict',
        messages: 'User exists on the platform',
        data: token
    }
}

export const badRequestResponse = {
    status: 400,
    response: {
        code: 400,
        status: 'Bad Request',
        messages: 'Invalid credentials',
        data: token
    }
}
