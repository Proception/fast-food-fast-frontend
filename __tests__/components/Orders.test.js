import React from 'react';
import { shallow } from 'enzyme';
import { Orders } from '../../src/components/Orders.jsx';
import {props, emptyOrderProps} from '../../__mocks__/orderCardsMock';

let wrapper;

describe('Orders Component', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<Orders { ...props }/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot', () => {
    wrapper = shallow(<Orders { ...emptyOrderProps }/>);
    expect(wrapper).toMatchSnapshot();
  });
});
