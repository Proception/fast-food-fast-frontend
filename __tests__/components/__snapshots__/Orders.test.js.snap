// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders Component should match snapshot 1`] = `
<div>
  <OrderControls />
  <section
    class="order"
  >
    <div
      class="container"
      id="sec"
    >
      <OrderCards
        orders={
          Array [
            Object {
              "id": 1,
              "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
              "orderDate": "30 August, 2016",
              "orderNo": "12",
              "orderStatus": "Pending",
              "price": 5000,
            },
            Object {
              "id": 2,
              "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
              "orderDate": "Boiled Yam",
              "orderNo": "54",
              "orderStatus": "In Progress",
              "price": 1200,
            },
          ]
        }
      />
      <div
        class="pagination"
      >
        <a
          href="#"
        >
          «
        </a>
        <a
          class="active"
          href="#"
        >
          1
        </a>
        <a
          href="#"
        >
          2
        </a>
        <a
          href="#"
        >
          »
        </a>
      </div>
    </div>
  </section>
</div>
`;

exports[`Orders Component should match snapshot 2`] = `
<div>
  <OrderControls />
  <section
    class="order"
  >
    <div
      class="container"
      id="sec"
    >
      <div
        class="new-order"
      >
        <div
          class="order-details"
        >
          There are no Orders
        </div>
      </div>
      <div
        class="pagination"
      >
        <a
          href="#"
        >
          «
        </a>
        <a
          class="active"
          href="#"
        >
          1
        </a>
        <a
          href="#"
        >
          2
        </a>
        <a
          href="#"
        >
          »
        </a>
      </div>
    </div>
  </section>
</div>
`;
